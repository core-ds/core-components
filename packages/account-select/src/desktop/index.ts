export { AccountSelectDesktop } from './Component.desktop';
