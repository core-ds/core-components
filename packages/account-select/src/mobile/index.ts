export { AccountSelectMobile } from './Component.mobile';
