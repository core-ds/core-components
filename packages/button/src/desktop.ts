export { ButtonDesktop } from './Component.desktop';
export type { CommonButtonProps as ButtonDesktopProps } from './typings';
