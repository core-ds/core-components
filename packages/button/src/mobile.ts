export { ButtonMobile } from './Component.mobile';
export type { CommonButtonProps as ButtonMobileProps } from './typings';
