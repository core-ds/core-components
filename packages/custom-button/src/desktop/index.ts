export { CustomButtonDesktop } from './Component.desktop';
