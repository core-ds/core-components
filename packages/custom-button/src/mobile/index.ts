export { CustomButtonMobile } from './Component.mobile';
