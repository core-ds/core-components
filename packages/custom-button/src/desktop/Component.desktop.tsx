import React, { forwardRef } from 'react';

import { ButtonDesktop } from '@alfalab/core-components-button/desktop';

import { BaseCustomButton } from '../components/base-custom-button';
import { CustomButtonProps } from '../types/props';

export const CustomButtonDesktop = forwardRef<
    HTMLAnchorElement | HTMLButtonElement,
    CustomButtonProps
>(({ children, ...restProps }, ref) => (
    <BaseCustomButton ref={ref} {...restProps} componentButton={ButtonDesktop}>
        {children}
    </BaseCustomButton>
));

CustomButtonDesktop.displayName = 'CustomButtonDesktop';
