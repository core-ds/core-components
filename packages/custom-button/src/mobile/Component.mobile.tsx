import React, { forwardRef } from 'react';

import { ButtonMobile } from '@alfalab/core-components-button/mobile';

import { BaseCustomButton } from '../components/base-custom-button';
import { CustomButtonProps } from '../types/props';

export const CustomButtonMobile = forwardRef<
    HTMLAnchorElement | HTMLButtonElement,
    CustomButtonProps
>(({ children, ...restProps }, ref) => (
    <BaseCustomButton ref={ref} {...restProps} componentButton={ButtonMobile}>
        {children}
    </BaseCustomButton>
));

CustomButtonMobile.displayName = 'CustomButtonMobile';
