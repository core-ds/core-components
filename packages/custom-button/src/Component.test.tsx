import React from 'react';
import { render } from '@testing-library/react';

import { CustomButton } from '.';
import { CustomButtonDesktop } from './desktop';
import { CustomButtonMobile } from './mobile';

const dataTestId = 'test-id';

describe('CustomButton', () => {
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation((query) => ({
            matches: true,
            media: query,
            onchange: null,
            addListener: jest.fn(), // Deprecated
            removeListener: jest.fn(), // Deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });

    describe('Snapshots tests', () => {
        it('should match snapshot', () => {
            expect(render(<CustomButton />)).toMatchSnapshot();
            expect(render(<CustomButtonDesktop />)).toMatchSnapshot();
            expect(render(<CustomButtonMobile />)).toMatchSnapshot();
        });

        it('should render custom background color', () => {
            expect(render(<CustomButton backgroundColor='#00ff00' />)).toMatchSnapshot();
            expect(render(<CustomButtonDesktop backgroundColor='#00ff00' />)).toMatchSnapshot();
            expect(render(<CustomButtonMobile backgroundColor='#00ff00' />)).toMatchSnapshot();
        });

        it('should render black color content', () => {
            expect(render(<CustomButton contentColor='black' />)).toMatchSnapshot();
            expect(render(<CustomButtonDesktop contentColor='black' />)).toMatchSnapshot();
            expect(render(<CustomButtonMobile contentColor='black' />)).toMatchSnapshot();
        });

        it('should render CustomButton view=primary by default', () => {
            expect(render(<CustomButton />)).toMatchSnapshot();
            expect(render(<CustomButtonDesktop />)).toMatchSnapshot();
            expect(render(<CustomButtonMobile />)).toMatchSnapshot();
        });
        it('should render loader if loading pass', () => {
            expect(render(<CustomButton loading={true} />)).toMatchSnapshot();
            expect(render(<CustomButtonDesktop loading={true} />)).toMatchSnapshot();
            expect(render(<CustomButtonMobile loading={true} />)).toMatchSnapshot();
        });
    });

    describe('Attributes tests', () => {
        it('should set `data-test-id` attribute', () => {
            const { getByTestId } = render(<CustomButton dataTestId={dataTestId} />);
            expect(getByTestId(dataTestId).tagName).toBe('BUTTON');
        });

        it('should have `style` attribute', () => {
            const { container } = render(<CustomButton />);
            expect(container.firstElementChild).toHaveAttribute('style');
        });

        it('should have `style background`', () => {
            const bgcolor = '#888';
            const { getByTestId } = render(
                <CustomButton dataTestId={dataTestId} backgroundColor={bgcolor} />,
            );
            expect(getByTestId(dataTestId)).toHaveStyle({ background: bgcolor });
        });

        it('should set type attribute', () => {
            const type = 'submit';
            const { container } = render(<CustomButton type={type} />);
            expect(container.firstElementChild).toHaveAttribute('type', type);
        });
    });

    describe('Classes tests', () => {
        it('should set `className` class', () => {
            const className = 'test-class';
            const { container } = render(<CustomButton className={className} />);

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should have `customButton` class as default', () => {
            const { container } = render(<CustomButton />);
            expect(container.firstElementChild).toHaveClass('customButton');
        });

        it('should have `white` class as default', () => {
            const { container } = render(<CustomButton />);
            expect(container.firstElementChild).toHaveClass('customButton');
        });

        it('should have `black` class', () => {
            const { container } = render(<CustomButton contentColor='black' />);
            expect(container.firstElementChild).toHaveClass('black');
        });

        describe('disableType test', () => {
            it('should have `disableType-default` class', () => {
                const { container } = render(<CustomButton disableType='default' />);
                expect(container.firstElementChild).toHaveClass('disableType-default');
            });

            it('should have `disableType-static` class', () => {
                const { container } = render(<CustomButton disableType='static' />);
                expect(container.firstElementChild).toHaveClass('disableType-static');
            });

            it('should have `disableType-inverted` class', () => {
                const { container } = render(<CustomButton disableType='inverted' />);
                expect(container.firstElementChild).toHaveClass('disableType-inverted');
            });

            it('should have `disableType-static-inverted` class', () => {
                const { container } = render(<CustomButton disableType='static-inverted' />);
                expect(container.firstElementChild).toHaveClass('disableType-static-inverted');
            });
        });

        it('should have custom className', () => {
            const { container } = render(<CustomButton className='customClass' />);
            expect(container.firstElementChild).toHaveClass('customClass');
        });
    });

    it('should unmount without errors', () => {
        const { unmount } = render(
            <CustomButton leftAddons={<span>Left</span>} rightAddons={<span>Right</span>}>
                Text
            </CustomButton>,
        );

        expect(unmount).not.toThrowError();
    });
});
