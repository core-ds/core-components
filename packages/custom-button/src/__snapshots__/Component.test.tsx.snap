// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton Snapshots tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="component primary m component primary iconOnly customButton white"
        style="background: rgb(255, 69, 195);"
        type="button"
      />
    </div>
  </body>,
  "container": <div>
    <button
      class="component primary m component primary iconOnly customButton white"
      style="background: rgb(255, 69, 195);"
      type="button"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CustomButton Snapshots tests should render CustomButton view=primary by default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="component primary m component primary iconOnly customButton white"
        style="background: rgb(255, 69, 195);"
        type="button"
      />
    </div>
  </body>,
  "container": <div>
    <button
      class="component primary m component primary iconOnly customButton white"
      style="background: rgb(255, 69, 195);"
      type="button"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CustomButton Snapshots tests should render black color content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="component primary m component primary iconOnly customButton black"
        style="background: rgb(255, 69, 195);"
        type="button"
      />
    </div>
  </body>,
  "container": <div>
    <button
      class="component primary m component primary iconOnly customButton black"
      style="background: rgb(255, 69, 195);"
      type="button"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CustomButton Snapshots tests should render custom background color 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="component primary m component primary iconOnly customButton white"
        style="background: rgb(0, 255, 0);"
        type="button"
      />
    </div>
  </body>,
  "container": <div>
    <button
      class="component primary m component primary iconOnly customButton white"
      style="background: rgb(0, 255, 0);"
      type="button"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CustomButton Snapshots tests should render loader if loading pass 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="component primary m component primary iconOnly loading customButton white customLoading"
        disabled=""
        style="background: rgb(255, 69, 195);"
        type="button"
      >
        <svg
          class="spinner component s loader loader visible"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <lineargradient
              id=":r0:_top"
              x1="0.05"
            >
              <stop
                offset="0.1"
                stop-color="currentColor"
                stop-opacity="0"
              />
              <stop
                offset="1"
                stop-color="currentColor"
                stop-opacity="0.3"
              />
            </lineargradient>
            <lineargradient
              id=":r0:_bottom"
              x1="0.05"
            >
              <stop
                offset="0"
                stop-color="currentColor"
                stop-opacity="1"
              />
              <stop
                offset="1"
                stop-color="currentColor"
                stop-opacity="0.3"
              />
            </lineargradient>
          </defs>
          <g
            stroke-width="2"
          >
            <path
              d="M3,12 A9,9 0 0 1 21,12"
              stroke="url(#:r0:_top)"
            />
            <path
              d="M21,12 A9,9 0 0 1 3,12"
              stroke="url(#:r0:_bottom)"
            />
            <path
              d="M3,12 A9,9 0 0 1 3 12"
              stroke="currentColor"
              stroke-linecap="round"
            />
          </g>
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="component primary m component primary iconOnly loading customButton white customLoading"
      disabled=""
      style="background: rgb(255, 69, 195);"
      type="button"
    >
      <svg
        class="spinner component s loader loader visible"
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            id=":r0:_top"
            x1="0.05"
          >
            <stop
              offset="0.1"
              stop-color="currentColor"
              stop-opacity="0"
            />
            <stop
              offset="1"
              stop-color="currentColor"
              stop-opacity="0.3"
            />
          </lineargradient>
          <lineargradient
            id=":r0:_bottom"
            x1="0.05"
          >
            <stop
              offset="0"
              stop-color="currentColor"
              stop-opacity="1"
            />
            <stop
              offset="1"
              stop-color="currentColor"
              stop-opacity="0.3"
            />
          </lineargradient>
        </defs>
        <g
          stroke-width="2"
        >
          <path
            d="M3,12 A9,9 0 0 1 21,12"
            stroke="url(#:r0:_top)"
          />
          <path
            d="M21,12 A9,9 0 0 1 3,12"
            stroke="url(#:r0:_bottom)"
          />
          <path
            d="M3,12 A9,9 0 0 1 3 12"
            stroke="currentColor"
            stroke-linecap="round"
          />
        </g>
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
